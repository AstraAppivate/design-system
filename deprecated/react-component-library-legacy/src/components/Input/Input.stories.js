import React from 'react'
import { withKnobs } from '@storybook/addon-knobs'
import { storiesOf } from '@storybook/react'

import State from '@royalnavy/storybook-react-input-state'

import Input from './index'

const stories = storiesOf('Input', module)
stories.addDecorator(withKnobs)
stories.addParameters({
  info: { inline: true, header: false, propTablesExclude: [State] },
})

stories.add('Text', () => (
  <State>
    <Input
      name="user"
      label="User"
      type="text"
      value="fred"
      onChange={() => {}}
    />
    <Input
      name="colour"
      label="Colour"
      type="text"
      value=""
      onChange={() => {}}
    />
  </State>
))

stories.add('TextArea', () => (
  <State>
    <Input
      name="description"
      label="Description"
      type="textarea"
      value=""
      onChange={() => {}}
    />
  </State>
))

stories.add('Tel', () => (
  <State>
    <Input
      name="number"
      label="Number"
      type="tel"
      value=""
      onChange={() => {}}
    />
  </State>
))
