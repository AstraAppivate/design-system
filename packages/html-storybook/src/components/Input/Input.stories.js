import React from 'react'
import { withKnobs } from '@storybook/addon-knobs'
import { storiesOf } from '@storybook/react'

const stories = storiesOf('Input', module)
stories.addDecorator(withKnobs)

stories.add('Text', () => (
  <div class="rn-inputBlock">
    <label class="inputLabel" for="802a3a0e-b9b7-483a-9e5d-d33d753cb890">
      User
    </label>
    <input
      class="input hasContent"
      id="802a3a0e-b9b7-483a-9e5d-d33d753cb890"
      name="user"
      value="fred"
    />
  </div>
))

stories.add('TextArea', () => (
  <div class="rn-inputBlock">
    <label class="inputLabel" for="802a3a0e-b9b7-483a-9e5d-d33d753cb890">
      Description
    </label>
    <textarea
      class="textarea "
      id="802a3a0e-b9b7-483a-9e5d-d33d753cb890"
      rows="5"
      cols="80"
      name="description"
    />
  </div>
))

stories.add('Tel', () => (
  <div class="rn-inputBlock">
    <label class="inputLabel" for="802a3a0e-b9b7-483a-9e5d-d33d753cb890">
      Number
    </label>
    <input
      class="input "
      id="802a3a0e-b9b7-483a-9e5d-d33d753cb890"
      name="number"
      value=""
    />
  </div>
))
