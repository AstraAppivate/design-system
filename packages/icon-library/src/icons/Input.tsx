import React, { SVGProps } from 'react'

const SvgInput = (props: SVGProps<SVGSVGElement>) => (
  <svg width={16} height={16} {...props}>
    <defs>
      <path
        id="input_svg__a"
        d="M14 2.007H2c-.733 0-1.333.6-1.333 1.333V6H2V3.327h12v9.353H2V10H.667v2.673c0 .734.6 1.32 1.333 1.32h12c.733 0 1.333-.586 1.333-1.32V3.34c0-.74-.6-1.333-1.333-1.333zm-6.667 8.66L10 8 7.333 5.333v2H.667v1.334h6.666v2z"
      />
    </defs>
    <g fill="none" fillRule="evenodd">
      <mask id="input_svg__b" fill="#fff">
        <use xlinkHref="#input_svg__a" />
      </mask>
      <g fill="CurrentColor" mask="url(#input_svg__b)">
        <path d="M0 0h16v16H0z" />
      </g>
    </g>
  </svg>
)

export default SvgInput
