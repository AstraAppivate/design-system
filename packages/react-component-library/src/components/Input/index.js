import PropTypes from 'prop-types'
import React from 'react'
import uuid from 'uuid'

/**
 * The input component is a generic styled input component that
 * can be used hand in hand with redux forms. The component creates a form
 * control complete with label and icon, if specified.
 */
const Input = ({
  cols,
  id,
  label,
  onChange,
  name,
  rows,
  type,
  value,
  ...rest
}) => (
  <div className="rn-inputBlock">
    <label className="inputLabel" htmlFor={id}>
      {label}
    </label>
    {type === 'textarea' ? (
      <textarea
        className={`textarea ${value && value.length ? 'hasContent' : ''}`}
        id={id}
        value={value}
        rows={rows}
        cols={cols}
        name={name}
        onChange={onChange}
        {...rest}
      />
    ) : (
      <input
        className={`input ${value && value.length ? 'hasContent' : ''}`}
        id={id}
        value={value}
        name={name}
        onChange={onChange}
        {...rest}
      />
    )}
  </div>
)

Input.propTypes = {
  /**
   * If the type is a textarea the cols property is used to set the
   * number of columns to display (characters wide)
   */
  cols: PropTypes.string,
  /**
   * Optional ID that will be used for the input component, allows a
   * label to be linked to an input. If one is not provided a unique
   * id will be generated and used
   */
  id: PropTypes.string,
  /**
   * The label to be used with the input. All inputs require an input
   */
  label: PropTypes.string.isRequired,
  /**
   * The name to use for the input element.
   */
  name: PropTypes.string.isRequired,
  /**
   * When a function to handle changes to the value, normally providfed by
   * redux forms
   */
  onChange: PropTypes.func.isRequired,
  /**
   * If the type is a textarea the cols property is used to set the
   * number of columns to display (characters wide)
   */
  rows: PropTypes.string,
  /**
   * The type attribute to use in the input element. If 'textarea' is specified
   * then a textaarea element is used.
   */
  type: PropTypes.string.isRequired,
  /**
   * The value for the input
   */
  value: PropTypes.string.isRequired,
}

Input.defaultProps = {
  cols: '80',
  rows: '5',
  id: uuid(),
}

export default Input
