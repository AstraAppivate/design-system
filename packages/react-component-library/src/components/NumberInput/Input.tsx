import React from 'react'
import { isFinite, isNil } from 'lodash'
import { selectors } from '@royalnavy/design-tokens'
import styled, { css } from 'styled-components'

import { NumberInputUnit } from './NumberInputUnit'
import { UnitPosition } from './NumberInput'
import { useInputText } from './useInputText'
import { InputValidationProps } from '../../common/InputValidationProps'

const { color, fontSize, spacing } = selectors

interface InputProps extends InputValidationProps {
  hasFocus: boolean
  isDisabled?: boolean
  id?: string
  isCondensed: boolean
  label?: string
  name: string
  onBlur: (event: React.FormEvent<HTMLInputElement>) => void
  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void
  onFocus: (event: React.FormEvent<HTMLInputElement>) => void
  placeholder?: string
  unit: string
  unitPosition: UnitPosition
  value?: number
}

interface StyledInputProps {
  $hasLabel: boolean
  $isCondensed: boolean
  $offset: number
}

interface StyledLabelProps {
  $hasContent: boolean
  $hasFocus: boolean
  $hasUnit: boolean
}

const StyledInputWrapper = styled.div`
  position: relative;
  flex-grow: 1;
`

const StyledLabel = styled.label<StyledLabelProps>`
  display: block;
  z-index: 1;
  position: absolute;
  top: 0;
  left: 0;
  transform-origin: top left;
  transform: translate(${spacing('6')}, ${spacing('6')}) scale(1);
  transition: color 200ms cubic-bezier(0, 0, 0.2, 1) 0ms,
    transform 200ms cubic-bezier(0, 0, 0.2, 1) 0ms;
  pointer-events: none;
  color: ${color('neutral', '400')};
  font-size: ${fontSize('base')};

  ${({ $hasContent, $hasFocus, $hasUnit }) =>
    ($hasContent || $hasFocus || $hasUnit) &&
    css`
      transform: translate(${spacing('6')}, 6px) scale(0.8);
    `}
`

const StyledInput = styled.input<StyledInputProps>`
  display: block;
  box-sizing: border-box;
  width: 100%;
  margin: 0 0 0 ${({ $offset }) => `${$offset}px`};
  padding: ${({ $hasLabel, $isCondensed }) => {
    if ($isCondensed) {
      return spacing('3')
    }

    if ($hasLabel) {
      return `${spacing('10')} ${spacing('6')} ${spacing('2')}`
    }

    return spacing('6')
  }};
  border: 0;
  background: none;
  -webkit-tap-highlight-color: transparent;
  font-size: ${fontSize('base')};

  &:focus {
    outline: 0;
  }
`

export const Input: React.FC<InputProps> = ({
  hasFocus,
  isDisabled = false,
  id,
  isCondensed,
  label,
  placeholder = '',
  unit,
  unitPosition,
  value,
  ...rest
}) => {
  const hasLabel = !!(label && label.length)
  const { canShow, inputOffset, inputRef, unitOffset } = useInputText(
    value,
    unitPosition
  )

  return (
    <StyledInputWrapper>
      {hasLabel && (
        <StyledLabel
          $hasContent={!isNil(value)}
          $hasFocus={hasFocus}
          $hasUnit={!isNil(unit)}
          data-testid="number-input-label"
          htmlFor={id}
        >
          {label}
        </StyledLabel>
      )}

      {canShow && (
        <NumberInputUnit
          left={`${unitOffset}px`}
          top={hasLabel && spacing('4')}
        >
          {unit}
        </NumberInputUnit>
      )}

      <StyledInput
        $hasLabel={hasLabel}
        $isCondensed={isCondensed}
        $offset={inputOffset}
        data-testid="number-input-input"
        disabled={isDisabled}
        id={id}
        placeholder={placeholder}
        ref={inputRef}
        type="text"
        value={isFinite(value) && canShow ? value : ''}
        {...rest}
      />
    </StyledInputWrapper>
  )
}

Input.displayName = 'Input'
