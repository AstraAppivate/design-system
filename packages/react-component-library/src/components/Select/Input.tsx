import React from 'react'
import get from 'lodash/get'
import { components, InputProps } from 'react-select'
import { selectors } from '@royalnavy/design-tokens'
import styled from 'styled-components'

const { color, fontSize, spacing } = selectors

export interface SelectInputProps extends InputProps {
  'aria-label'?: string
  id?: string
}

const StyledInputContainer = styled.div`
  display: flex;
  align-items: center;
  height: inherit;

  &&& > * {
    padding: 0;
    margin: 0;
  }
`

export const StyledLabel = styled.label`
  display: block;
  font-size: ${fontSize('base')};
  color: ${color('neutral', '400')};
  position: absolute;
  top: 0;
  left: 0;
  transform-origin: top left;
  transform: translate(${spacing('6')}, ${spacing('6')}) scale(1);
  transition: color 200ms cubic-bezier(0, 0, 0.2, 1) 0ms,
    transform 200ms cubic-bezier(0, 0, 0.2, 1) 0ms;
`

const StyledInput = styled(components.Input)`
  font-size: ${fontSize('base')};
  margin: ${spacing('6')} 0 0;
`

export const Input: React.FC<SelectInputProps> = (props) => {
  const { 'aria-label': ariaLabel, ...rest } = props
  const inputTestId = get(
    props,
    'selectProps.data-testid',
    'react-select-vendor-input'
  )

  return (
    <StyledInputContainer>
      {ariaLabel && (
        <StyledLabel htmlFor={props.id} data-testid="select-label">
          {ariaLabel}
        </StyledLabel>
      )}
      <StyledInput data-testid={inputTestId} {...rest} />
    </StyledInputContainer>
  )
}
