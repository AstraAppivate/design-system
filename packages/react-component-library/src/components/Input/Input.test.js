import React from 'react'
import { shallow } from 'enzyme'

import Input from './index'

describe('Input', () => {
  let onChangeStub
  let wrapper

  beforeAll(() => {
    onChangeStub = jest.fn()
  })

  afterEach(() => {
    onChangeStub.mockReset()
  })

  describe('text', () => {
    beforeEach(() => {
      wrapper = shallow(
        <Input
          id="user"
          type="text"
          label="test"
          name="user"
          value="1234"
          onChange={onChangeStub}
        />
      )
    })

    it('should render correctly render the input component', () => {
      expect(wrapper).toMatchSnapshot()
    })

    it('should call the change handler when the user inputs a value', () => {
      const userInput = wrapper.find('input[name="user"]')
      userInput.value = 'correctUsername'
      userInput.simulate('change')
      expect(onChangeStub).toHaveBeenCalled()
    })
  })

  describe('textarea', () => {
    beforeEach(() => {
      wrapper = shallow(
        <Input
          id="user"
          type="textarea"
          label="test"
          name="user"
          value="1234"
          onChange={onChangeStub}
        />
      )
    })

    it('should render correctly render the input component', () => {
      expect(wrapper).toMatchSnapshot()
    })

    it('should call the change handler when the user inputs a value', () => {
      const userInput = wrapper.find('textarea[name="user"]')
      userInput.value = 'correctUsername'
      userInput.simulate('change')
      expect(onChangeStub).toHaveBeenCalled()
    })
  })
})
