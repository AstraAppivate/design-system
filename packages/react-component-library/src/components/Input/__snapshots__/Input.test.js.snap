// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input text should render correctly render the input component 1`] = `
<div
  className="rn-inputBlock"
>
  <input
    className="input hasContent"
    id="user"
    name="user"
    onChange={[MockFunction]}
    value="1234"
  />
  <label
    className="inputLabel"
    htmlFor="user"
  >
    <span
      className="inputLabelInner"
    >
      test
    </span>
  </label>
</div>
`;

exports[`Input textarea should render correctly render the input component 1`] = `
<div
  className="rn-inputBlock"
>
  <textarea
    className="textarea hasContent"
    cols="80"
    id="user"
    name="user"
    onChange={[MockFunction]}
    rows="5"
    value="1234"
  />
  <label
    className="inputLabel"
    htmlFor="user"
  >
    <span
      className="inputLabelInner"
    >
      test
    </span>
  </label>
</div>
`;
